src/game/stage/stage.c

```diff
@@ -623,26 +690,81 @@ void HandleDeath(void)
     }
 }
 
-// TODO:
 // Unused.
-// Might be a leftover from the first game?
+// Leftover from SA1, but smaller than in there.
 void GoToNextLevel(void)
 {
+    u16 irqEnable, irqMasterEnable, dispStat;
+
     TasksDestroyAll();
     PAUSE_BACKGROUNDS_QUEUE();
-    gUnknown_03005390 = 0;
+    SA2_LABEL(gUnknown_03005390) = 0;
     PAUSE_GRAPHICS_QUEUE();
+
+#if (GAME == GAME_SA1)
+    m4aMPlayAllStop();
+    m4aSoundVSyncOff();
+
+    gFlags |= FLAGS_8000;
+    irqEnable = REG_IE;
+    irqMasterEnable = REG_IME;
+    dispStat = REG_DISPSTAT;
+
+    REG_IE = 0;
+    REG_IE;
+    REG_IME = 0;
+    REG_IME;
+    REG_DISPSTAT = 0;
+    REG_DISPSTAT;
+
+    gFlags &= ~FLAGS_4;
+
+    SlowDmaStop(0);
+    SlowDmaStop(1);
+    SlowDmaStop(2);
+    SlowDmaStop(3);
```

Comment:
This is a macro we could make? Ive seen it used elsewhere

Reply:
Maybe in a different PR?

Something that makes this less feasible is `SlowDmaStop`, but we don't yet know how of the Slow variant was used.